pkg/build/api/types.go

```diff
@@ -25,6 +25,8 @@ const (
 	BuildJenkinsLogURLAnnotation = "openshift.io/jenkins-log-url"
 	// BuildJenkinsBuildURIAnnotation is an annotation holding a link to the Jenkins build
 	BuildJenkinsBuildURIAnnotation = "openshift.io/jenkins-build-uri"
+	// BuildMatchSourceURIAnnotation is an annotation on a Secret indicating the URIs for which it can be used
+	BuildMatchSourceURIAnnotation = "build.openshift.io/match-source-uri"
```

Comment:
>  I don't think openshift.io/ is reasonable when your type is in the build.openshift.io group. You probably need to stop writing the old ones and start transitioning.

Well, for what it's worth (nothing), personally i don't this is worth renaming at all.  we really need to start living with our choices(that i should note, have mostly been api-review approved at some point previously) instead of churning for dubious improvements that leads to us having to maintain spaghetti paths of code to support previous(and now deemed wrong) choices.

Just a quick search of all the places these sort of things need to change if we're picking a new convention:
```
	SupplementalGroupsAnnotation = "openshift.io/sa.scc.supplemental-groups"
	MCSAnnotation                = "openshift.io/sa.scc.mcs"
	ValidatedSCCAnnotation       = "openshift.io/scc"
	DeploymentConfigAnnotation = "openshift.io/deployment-config.name"
	DeploymentAnnotation = "openshift.io/deployment.name"
	DeploymentPodAnnotation = "openshift.io/deployer-pod.name"
	DeploymentStatusAnnotation = "openshift.io/deployment.phase"
	DeploymentEncodedConfigAnnotation = "openshift.io/encoded-deployment-config"
	DeploymentVersionAnnotation = "openshift.io/deployment-config.latest-version"
	DeploymentStatusReasonAnnotation = "openshift.io/deployment.status-reason"
	DeploymentCancelledAnnotation = "openshift.io/deployment.cancelled"
	DeploymentReplicasAnnotation = "openshift.io/deployment.replicas"
```

(there are many more)

Not to mention the UI will also need to look for both values in a lot of these cases.

I think we need a better justification for such a migration.